### ISSUE TITLE
Consider alternatives to reflection 

### ISSUE THREAD

**Issue by ryancrawcour** [ryancrawcour/cosmosdb-graph-bulkexecutor#2]
The reflection approach, while quicker to bang out, is going to be quite costly on a per call basis. If you switched to building out expressions trees and baking/caching delegates you could pay those costs one time per type and eliminate almost all ongoing runtime costs on subsequent calls.

_Originally posted by @drub0y in https://github.com/ryancrawcour/cosmosdb-graph-bulkexecutor/pull/1#issuecomment-542396034_

**Comment by hannesne** [comment]
Expression trees is used to read the dynamic values in #7

**Comment by ryancrawcour** [comment]
Thanks. That's only for Dynamic objects, right? The rest still uses Reflection?

**Comment by hannesne** [comment]
Yeah only dynamic. I didn't change the approach for regular reads. I'm not sure if using expression trees will be vastly different, as you still need to use the reflection component to determine the name of the property you want to read.
